frontend/hub/collections/CollectionPage/CollectionContents.tsx

```diff
@@ -1,7 +1,226 @@
+    pulpAPI`/content/ansible/collection_versions/?name=${
+      collection.collection_version?.name || ''
+    }&namespace=${collection.collection_version?.namespace || ''}&version=${
+      collection.collection_version?.version || ''
+    }`
+  );
+
+  useEffect(() => {
+    // focus on keywords filter change - it will anyway loose focus
+    searchRef?.current?.focus();
+  }, [keywords]);
+
+  const contents = data?.results[0].contents || [];
+  const summary: Record<string, number> = { all: 0 };
+
+  const toShow: CollectionContent[] = [];
+
+  if (contents.length > 0) {
+    for (const c of contents) {
+      summary[c.content_type] ||= 0;
+
+      const keywordMatch = c.name.match(keywords);
+      const typeMatch = showing === 'all' ? true : c.content_type === showing;
+
+      // count only items matching keyword
+      if (keywordMatch) {
+        summary[c.content_type]++;
+        summary['all']++;
+      }
+
+      // show only items matching keyword + type
+      if (keywordMatch && typeMatch) {
+        toShow.push(c);
+      }
+    }
+  }
+
+  const label_style = `
+    margin-right: 25px;
+    text-transform: capitalize;
+    font-size: 16px;
+  `;
+
+  const StyledDiv = styled.div`
+    .hub-c-toolbar__item-type-selector {
+      ${label_style}
+    }
+
+    .hub-c-toolbar__item-type-selector-label {
+      ${label_style}
+    }
+
+    .hub-c-toolbar__item-selected-item,
+    .hub-c-toolbar__item-type-selector:hover {
+      border-bottom: 2px solid var(--pf-global--link--Color);
+      padding-bottom: 2px;
+      margin-bottom: -4px;
+      background-color: white;
+      cursor: pointer;
+      text-decoration: underline;
+      text-decoration-color: blue; /* Sets the underline color to blue */
+      text-decoration-thickness: 2px; /* Sets the thickness of the underline */
+      text-underline-offset: 8px; /* Sets the space between the text and the underline */
+    }
+  `;
+
+  if (error || data?.results?.length === 0) {
+    return <HubError error={error} handleRefresh={refresh} />;
+  }
+
+  if (!data && !error) {
+    return <LoadingPage />;
+  }
+
+  return (
+    <>
+      <PageSection variant="light">
+        <StyledDiv>
+          <SearchInput
+            key="content-search-input-key"
+            id="content-search-input-id"
+            placeholder={t('Find content')}
+            value={keywords}
+            ref={searchRef}
+            onChange={(value) => {
+              setSearchParams((params) => {
+                params.set('keywords', value?.currentTarget?.value || '');
+                return params;
+              });
+            }}
+          />
+          <br />
+          <span className={'hub-c-toolbar__item-type-selector-label'}>{t`Showing`}:</span>
+          {Object.keys(summary).map((key) => {
+            // eslint-disable-next-line i18next/no-literal-string
+            let className = 'clickable hub-c-toolbar__item-type-selector';
```

Comment:
I was not able to follow this one clear, could we create a custom styled component, and pass a prop to select what css should be applied.

Reply:
I dont know if I understand it well. So you want me to not use StyledDiv, but rather pass final style definitions (like color and such) without classes as css selectors? That may be better, truth to be told, I dont like css files, they tend to be very confusing, you never know what style will be applied and everything is defined in magic places...